src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/SqlParser.java

```diff
@@ -218,7 +219,7 @@ private void addOrderByToSelect(Select select, List<SQLSelectOrderByItem> items,
             Field field = FieldMaker.makeField(expr, null, null);
 
             String orderByName;
-            if (isScriptField(field)) {
+            if (field instanceof ScriptMethodField) {
```

Comment:
Since we've had a class/abstraction for script method field, I think we should move this to a new method in Field class hierachy like `getOrderbyName` to take advantage of polymorphism and avoid `instanceof` as much as possible (as you said, we've had too many).

Reply:
Done